src/lminiz.c

```diff
@@ -39,6 +39,14 @@ static size_t lmz_file_read(void *pOpaque, mz_uint64 file_ofs, void *pBuf, size_
   return zip->req.result;
 }
 
+static int lmz_check_compression_level(lua_State* L, int index) {
+  int level = luaL_optint(L, index, MZ_DEFAULT_COMPRESSION);
+  if (level < MZ_DEFAULT_COMPRESSION || level > MZ_BEST_COMPRESSION) {
```

Comment:
hmm. Shouldn't this be MZ_NO_COMPRESSION as the minimum?

Reply:
Maybe.

In miniz:

> Compression levels: 0-9 are the standard zlib-style levels, 10 is best possible compression (not zlib compatible, and may be very slow),

```c
enum
{
    MZ_NO_COMPRESSION = 0,
    MZ_BEST_SPEED = 1,
    MZ_BEST_COMPRESSION = 9,
    MZ_UBER_COMPRESSION = 10,
    MZ_DEFAULT_LEVEL = 6,
    MZ_DEFAULT_COMPRESSION = -1
};
```

The zlib manual says:
> The compression level must be Z_DEFAULT_COMPRESSION, or between 0 and 9: 1 gives best speed, 9 gives best compression, 0 gives no compression at all (the input data is simply copied a block at a time). Z_DEFAULT_COMPRESSION requests a default compromise between speed and compression (currently equivalent to level 6). 

No compression should be the minimum. Should we raise the maximum to uber compression?

Reply:
ahh... DEFAULT is -1... nevermind. +1